specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+  /// Read more about HTTP client certificates at
+  /// [RFC 8446 The Transport Layer Security (TLS) Protocol Version 1.3](https://tools.ietf.org/html/rfc8446).
+  ///
+  /// With this event you have several options for responding to client certificate requests:
+  ///
+  /// *  You can query the mutually trusted CA certificates, and select a certificate
+  /// from the list to respond to the server.
+  /// *  You can choose to respond to the server without a certificate.
+  /// *  You can choose to display default client certificate selection dialog prompt
+  /// to let user to respond to the server.
+  /// *  You can cancel the request.
+  ///
+  /// If you doesn't handle the event, WebView2 will
+  /// show the default client certificate selection dialog prompt to user.
+  ///
+  /// \snippet SettingsComponent.cpp ClientCertificateRequested
+  HRESULT add_ClientCertificateRequested(
+      [in] ICoreWebView2ClientCertificateRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_ClientCertificateRequested.
+  HRESULT remove_ClientCertificateRequested([in] EventRegistrationToken token);
+}
+
+/// Provides access to the certificate metadata
+[uuid(9ca02116-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2ClientCertificate : IUnknown {
+  /// Subject of the certificate.
+  [propget] HRESULT Subject([out, retval] LPWSTR* subject);
+  /// Name of the certificate authority that issued the certificate.
+  [propget] HRESULT Issuer([ out, retval ] LPWSTR* issuer);
+  /// The valid start date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidStartDate([out, retval] double* validStartDate);
+  /// The valid expiration date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidExpiryDate([out, retval] double* validExpiryDate);
+  /// DER encoded serial number of the certificate.
+  [propget] HRESULT SerialNumber([out, retval] LPWSTR* serialNumber);
+  /// Display name for a certificate.
+  [propget] HRESULT DisplayName([out, retval] LPWSTR* displayName);
+  /// PEM encoded data for the certificate.
+  /// Returns Base64 encoding of DER encoded certificate.
+  /// Read more about PEM at [RFC 1421 Privacy Enhanced Mail]
+  /// (https://tools.ietf.org/html/rfc1421).
+  [propget] HRESULT PEMEncodedData([out, retval] LPWSTR* pemEncodedData);
+  /// List of PEM encoded client certificate issuer chain.
+  /// This list contains the certificate and intermediate CA certificates.
+  [propget] HRESULT PEMEncodedIssuerChainData([out, retval]
+      ICoreWebView2ClientCertificateIssuerChainList**
+                                              pemEncodedIssuerChainData);
+  /// Kind of a certificate (smart card, pin).
+  [propget] HRESULT CertificateKind([out,retval]
```

Comment:
I think `Kind` is sufficient, since it's already a member of a Certificate object.